recurring_contract/model/contract_group.py

```diff
@@ -82,22 +91,102 @@ def _get_groups_from_contract(self, cr, uid, ids, context=None):
                     _get_groups_from_contract, ['next_invoice_date',
                                                 'state'], 20),
             }),
+        'last_paid_invoice_date': fields.function(
+            _get_last_paid_invoice, type='date',
+            string=_('Last paid invoice date')
+        )
     }
 
     _defaults = {
         'ref': '/',
         'recurring_unit': 'month',
         'recurring_value': 1,
+        'advance_billing_months': 1,
     }
 
+    def _get_contract_ids(self, cr, uid, ids, context=None):
+        contract_ids = list()
+
+        for contract_group_id in self.browse(cr, uid, ids, context):
+            for contract in contract_group_id.contract_ids:
+                if contract.id not in contract_ids:
+                    contract_ids.append(contract.id)
+        return contract_ids
+
     def write(self, cr, uid, ids, vals, context=None):
+
+        recurring_contract_obj = self.pool.get('recurring.contract')
+        contract_ids = self._get_contract_ids(cr, uid, ids, context)
+
+        if ('advance_billing_months' in vals):
+            old_advance_billing_months = self.browse(
+                cr, uid, ids, context)[0].advance_billing_months

```

Comment:
It's not a good practice to take into consideration only the first id. Make the method handle the case where several ids would be passed.